pkg/cpuallocator/allocator.go

```diff
@@ -943,6 +984,40 @@ func (c *cpuPriorities) cmpCPUSet(csetA, csetB cpuset.CPUSet, prefer CPUPriority
 		return 0
 	}
 
+	// For low prio request, favor cpuset with the tightest fit.
+	if cpuCnt > 0 && prefer == PriorityLow {
+		prefA := csetA.Intersection(c[prefer]).Size()
+		prefB := csetB.Intersection(c[prefer]).Size()
+		if prefA >= cpuCnt && prefB >= cpuCnt {
+			return prefA - prefB
```

Comment:
Wouldn't your orignal (or the like) code work just as expected?

```suggestion
		if prefA >= cpuCnt || prefB >= cpuCnt {
			return prefA - prefB
```

Reply:
Damn I got really confused here, making a mental association to `slices.SortFunc()` somehow... probably due to the return type.

Anyway, @marquiz no, that wouldn't work correctly. We need to take the smaller one which is larger than cpuCnt, keeping in mind that we need to return > 0 to prefer A and < 0 to prefer B.

Reply:
Well, then `return prefA - prefB`?

It should work. If BOTH `prefA >= cpuCnt` AND `prefB >= cpuCnt` then we just return prefA-prefB as you did.

If EITHER `prefA >= cpuCnt` OR `prefB >= cpuCnt` then we return either positive or negative number (because one of those is greater than the other), just like in your code. The value doesn't matter. It doesn't need to be -1 or 1.

Or is there something I'm missing here 🤔

Reply:
Hmm, wouldn't it work incorrectly, for instance, in this case:
cpuCnt: 2
prefA: 2
prefB: 3
prefA - prefB = 2 - 3 = -1 => we prefer B but we should prefer A as the tighter satisfying fit.

So we have to return prefB - prefA to work correctly, but only if both are greater than cpuCnt. Otherwise, wouldn't,for instance, this work incorrectly:
cpuCnt: 2
prefA: 2
prefB: 1
prefB - prefA = -1 => we prefer B, but we should prefer A, since B has insufficient CPUs.

Reply:
Argh, yeah you're right 🙈 Depending on the precondition we choose either the bigger or the smaller set, like

```golang

		if prefA >= cpuCnt && prefB >= cpuCnt {
			// Both sets can satisfy the req -> choose the smaller (tighter fit)
			return prefA - prefB
		} else if prefA >= cpuCnt || prefB >= cpuCnt {
			// Only one set is suitable -> we need to pick it (the bigger)
			return prefB - prefA
		}
```

Maybe this could be documented more explicitly in the code comment to avoid future confusion (at least from my part) (?)

Reply:
Hmm... but the `return prefB - prefA` vs. `return prefA - prefB` should be the other way around above...

Reply:
Updated along those lines and with comments to that effect.

Reply:
Ah yes, I had those wrong :/ Anyway, thanks for the update, easier to read for me, now